internal/commands/executor.go

```diff
@@ -2,23 +2,25 @@ package commands
 
 import (
 	"fmt"
-	"os"
 	"time"
 
+	"github.com/UpCloudLtd/progress"
+	"github.com/UpCloudLtd/progress/messages"
 	"github.com/UpCloudLtd/upcloud-cli/internal/config"
 	"github.com/UpCloudLtd/upcloud-cli/internal/output"
 	internal "github.com/UpCloudLtd/upcloud-cli/internal/service"
-	"github.com/UpCloudLtd/upcloud-cli/internal/ui"
 
 	"github.com/UpCloudLtd/upcloud-go-api/v4/upcloud/service"
 	"github.com/gemalto/flume"
 )
 
 // Executor represents the execution context for commands
 type Executor interface {
```

Comment:
Not totally related, but I think interface is not needed as we only seem to pass around `executorImp`. Would it make sense to remove the interface and rename the struct to `Executor` at this point?

Reply:
I think we could create a separate ticket for that as this is already quite big PR